csrc/flash_attn/flash_api.cpp

```diff
@@ -947,7 +951,7 @@ mha_varlen_bwd(const at::Tensor &dout,  // total_q x num_heads, x head_size
     at::cuda::CUDAGuard device_guard{(char)q.get_device()};
 
     auto opts = q.options();
-    auto softmax_d = torch::empty({batch_size, num_heads, seqlen_q_rounded}, opts.dtype(at::kFloat));
+    auto softmax_d = torch::empty({num_heads, total_q + 128 * batch_size}, opts.dtype(at::kFloat));
```

Comment:
same here, are we expecting this tensor to be on the device?
```suggestion
    auto softmax_d = torch::empty({num_heads, total_q + 128 * batch_size}, dtype=opts.dtype(at::kFloat), device=q.get_device());
```